scripts/joinmarket-qt.py

```diff
@@ -1160,13 +1165,14 @@ def validateSingleSend(self):
 different mixdepths.
 
 Are you sure you know what you're doing?"""
-            reply = JMQtMessageBox(self, msg, mbtype='question', title="Warning")
-            if reply == QMessageBox.No:
-                return False
+            if makercount > 0:
+                reply = JMQtMessageBox(self, msg, mbtype='question', title="Warning")
+                if reply == QMessageBox.No:
+                    return False
 
             change_addr_obj = btc.CCoinAddress(change_addr)
             if change_addr_obj.get_scriptPubKey_type() != "witness_v0_keyhash":
```

Comment:
It'll be better to make a check against the wallet. Some people are still using p2sh-p2wpkh and we can accommodate that here.

The wallet has a TYPE member var which is in `[TYPE_P2PKH, TYPE_P2WPKH, TYPE_P2SH_P2WPKH]`, with the first being pretty much unused. You can call `jmclient.detect_script_type(self.wallet_service.addr_to_script(address))` \* and compare the value with the aforementioned `wallet.TYPE`. It could be a little cleaner that it is but it's fine here. 

\* You'll have to export detect_script_type in the jmclient __init__.py, it's not exported yet.

Reply:
> Some people are still using p2sh-p2wpkh and we can accommodate that here.

Oh, good point. Actually not just accommodate--it's important that the "unusual address" detection works properly regardless of wallet type. As it is, wrapped SegWit users get the exact opposite of what should happen: No special warning when doing something unusual (beyond just using custom change), and a warning when not doing anything unusual.

> The wallet has a TYPE member var [...]

Thanks for the pointers; that's definitely the way to go.